modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java

```diff
@@ -134,6 +145,27 @@ else if (entity instanceof InputStreamHttpEntity)
                 new DefaultMuleSession());
     }
 
+    private static void resolveCorrelationId(DefaultMuleMessage defaultMuleMessage) {
+        String xCorrelationId = defaultMuleMessage.getInboundProperty(X_CORRELATION_ID);
+        String muleCorrelationId = defaultMuleMessage.getInboundProperty(MULE_CORRELATION_ID_PROPERTY);
+        if (muleCorrelationId != null)
+        {
+            if (xCorrelationId != null)
+            {
+                LOGGER.warn("'X-Correlation-ID' and 'MULE_CORRELATION_ID' headers found.");
+                if (!xCorrelationId.equals(muleCorrelationId))
+                {
+                    LOGGER.warn("Correlation values do not match. 'MULE_CORRELATION_ID' will be used.");
```

Comment:
will be hard for the user reading the log to correlate this message to the one above. Ideally each situation should have one complete message, which should also include the correlation values